### PR TITLE
pybind/mgr/restful: use list to pass hooks to create a `Pecan` instance

### PR DESCRIPTION
**Pull request by tchaikov**
pecan 0.3.2 introduced the new way to pass the hooks to construct a
Pecan instance, but the "callable" builder for hooks is not compatible
with pecan < 0.3.2. and ubuntu trusty ships pecan 0.3.0, so we need to
do this the old way: pass a list of hooks instead of a callable.

see https://github.com/pecan/pecan/blame/0.3.2/pecan/core.py and
https://github.com/pecan/pecan/commit/fb2cf166aaf5b5af461879ec5b7a46806e8aee45

Fixes: http://tracker.ceph.com/issues/20258
Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

**Comment by b-ranto** [comment]
@tchaikov can you run it through the rados test suite to see if it still works with pecan version in centos? The 0.3.2 source code suggests it should but you never know. Or even better, can you update the test case yaml file (` qa/suites/rados/rest/mgr-restful.yaml`) to run on trusty (and xenial) as well and run it through the rados test case?

**Comment by tchaikov** [comment]
@b-ranto sure, will do.

**Comment by tchaikov** [comment]
@b-ranto the test failed for a different reason: 

- http://qa-proxy.ceph.com/teuthology/kchai-2017-06-13_13:01:38-rados-wip-20258-kefu---basic-smithi/1282996/teuthology.log: (xenial) looks like a recursive call caused a stack overflow.
- http://qa-proxy.ceph.com/teuthology/kchai-2017-06-13_12:56:55-rados-wip-20258-kefu---basic-smithi/1282994/teuthology.log: (centos) we are expecting a json,

**Comment by tchaikov** [comment]
i will rerun the test with master tomorrow.

**Comment by tchaikov** [comment]
retest this please.

**Comment by tchaikov** [comment]
tested at 
- http://pulpito.ceph.com/kchai-2017-06-15_04:14:36-rados-wip-20258-kefu---basic-mira/
- http://pulpito.ceph.com/kchai-2017-06-15_04:14:01-rados-wip-20258-kefu---basic-mira/

**Comment by tchaikov** [comment]
@b-ranto @liewegas the tests pass now. could you take a look again?

File: src/pybind/mgr/restful/api/pool.py [hunk 1]
```diff
@@ -34,7 +34,11 @@ def patch(self, **kwargs):
         """
         Modify the information for the pool id
         """
-        args = request.json
+	try:
+            args = request.json
+        except ValueError:
+            response.status = 400
+            return {'message': 'Bad request'}
```
Comment (b-ranto) [inline src/pybind/mgr/restful/api/pool.py]: Can we be a bit more specific here and return something like 'Bad request: malformed json'?

**Comment by tchaikov** [comment]
@b-ranto updated and repushed.

**Pull request merged**